src/torManager/torManager.ts

```diff
@@ -107,28 +131,39 @@ export class Tor {
     return byPlatform[process.platform]
   }
 
-  protected readonly spawnTor = resolve => {
-    this.process = child_process.spawn(
-      this.torPath,
-      [
-        '--SocksPort',
-        this.socksPort,
-        ...(this.httpTunnelPort ? ['--HTTPTunnelPort', this.httpTunnelPort] : []),
-        '--ControlPort',
-        this.controlPort.toString(),
-        '--PidFile',
-        this.torPidPath,
-        '--DataDirectory',
-        this.torDataDirectory,
-        '--HashedControlPassword',
-        this.torHashedPassword
-      ],
-      this.options
-    )
-    this.process.stdout.on('data', data => {
-      log(data.toString())
-      const regexp = /Bootstrapped 100%/
-      if (regexp.test(data.toString())) resolve()
+  protected readonly spawnTor = async (timeoutMs): Promise<void> => {
+    return await new Promise((resolve, reject) => {
+      this.process = child_process.spawn(
+        this.torPath,
+        [
+          '--SocksPort',
+          this.socksPort,
+          ...(this.httpTunnelPort ? ['--HTTPTunnelPort', this.httpTunnelPort] : []),
+          '--ControlPort',
+          this.controlPort.toString(),
+          '--PidFile',
+          this.torPidPath,
+          '--DataDirectory',
+          this.torDataDirectory,
+          '--HashedControlPassword',
+          this.torHashedPassword
+        ],
+        this.options
+      )
+
+      const timeout = setTimeout(() => {
+        // eslint-disable-next-line
+        reject()
```

Comment:
reject(new Error('Timeout of ${timeoutMs / 1000} while waiting for tor to bootstrap`))